src/plugins/workspace/public/components/workspace_selector/workspace_selector.tsx

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import { useObservable } from 'react-use';
+import { i18n } from '@osd/i18n';
+import {
+  EuiButton,
+  EuiIcon,
+  EuiPopover,
+  EuiPanel,
+  EuiHorizontalRule,
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import { BehaviorSubject } from 'rxjs';
+import { WORKSPACE_CREATE_APP_ID, WORKSPACE_LIST_APP_ID } from '../../../common/constants';
+import { CoreStart, WorkspaceObject } from '../../../../../core/public';
+import { getFirstUseCaseOfFeatureConfigs } from '../../utils';
+import { WorkspaceUseCase } from '../../types';
+import { validateWorkspaceColor } from '../../../common/utils';
+import { WorkspacePickerContent } from '../workspace_picker_content/workspace_picker_content';
+
+const createWorkspaceButton = i18n.translate('workspace.menu.button.createWorkspace', {
+  defaultMessage: 'Create workspace',
+});
+
+const manageWorkspacesButton = i18n.translate('workspace.menu.button.manageWorkspaces', {
+  defaultMessage: 'Manage',
+});
+
+const getValidWorkspaceColor = (color?: string) =>
+  validateWorkspaceColor(color) ? color : undefined;
+
+interface Props {
+  coreStart: CoreStart;
+  registeredUseCases$: BehaviorSubject<WorkspaceUseCase[]>;
+}
+
+export const WorkspaceSelector = ({ coreStart, registeredUseCases$ }: Props) => {
+  const [isPopoverOpen, setPopover] = useState(false);
+  const currentWorkspace = useObservable(coreStart.workspaces.currentWorkspace$, null);
+  const availableUseCases = useObservable(registeredUseCases$, []);
+  const isDashboardAdmin = coreStart.application.capabilities?.dashboards?.isDashboardAdmin;
+
+  const getUseCase = (workspace: WorkspaceObject) => {
+    if (!workspace.features) {
+      return;
+    }
+    const useCaseId = getFirstUseCaseOfFeatureConfigs(workspace.features);
+    return availableUseCases.find((useCase) => useCase.id === useCaseId);
+  };
+
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setPopover(false);
+  };
+  const button = currentWorkspace ? (
+    <EuiPanel paddingSize="none" color="transparent" hasBorder={false} hasShadow={false}>
+      <EuiText
+        size="xs"
+        // here I try inline style to achive the label looks
+        style={{
+          position: 'relative',
+          bottom: '-10px',
+          zIndex: 1,
+          padding: '0 5px',
+        }}
+      >
+        <p>worksapce</p>
```

Comment:
It looks like should be `small` text
```suggestion
        <small>worksapce</small>
```